pkg/management/authorization.go

```diff
@@ -253,6 +254,17 @@ func (c *authorizationComponentMW) GetGroupsOfUser(ctx context.Context, realmNam
 	return c.next.GetGroupsOfUser(ctx, realmName, userID)
 }
 
+func (c *authorizationComponentMW) SetGroupsToUser(ctx context.Context, realmName, userID string, groupIDs []string) error {
+	var action = MGMTSetGroupsToUser.String()
+	var targetRealm = realmName
+
+	if err := c.authManager.CheckAuthorizationOnTargetUser(ctx, action, targetRealm, userID); err != nil {
```

Comment:
I had not anticipated it but we would need a way to filter the list of group which are assignable by someone. I would suggest to add a new action MGMTAssignableGroupsToUser and add another check in this method which would ensure the groupIDs in parameters are among the allowed ones. 
What do you think of it ?

Reply:
Code updated... please check once again